### PR TITLE
VR-5909 new download docker context CLI

### PR DESCRIPTION
**Pull request by evgeniilezhnin**

### CODE REVIEW

**Comment by evgeniilezhnin** [comment]
I didn't validate it yet, because there's error on backend now: https://vertaai.slack.com/archives/CJ2MM54Q3/p1597389855362300

File: client/verta/tests/test_cli/test_endpoint.py [hunk 1]
```diff
@@ -409,6 +410,45 @@ def test_update_autoscaling(self, client, created_endpoints, experiment_run, mod
...
+        # can be loaded as tgz
+        with tarfile.open(download_to_path, 'r:gz') as f:
+            filepaths = set(f.getnames())
+
+        assert "Dockerfile" in filepaths
+
+        runner = CliRunner()
+        result = runner.invoke(
+            cli,
+            ['deployment', 'download', 'dockercontext', '--model_version_id', model_version.id, '--output', download_to_path],
```
Comment (convoliution) [inline client/verta/tests/test_cli/test_endpoint.py]: For the model version test, I think we should use a new ModelVersion _not_ `from_run()`.
So, using `model_version.log_model()`, `log_artifact()`, and `log_environment()`

File: client/verta/verta/_cli/deployment/download.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+            entity = client.get_experiment_run(id=run_id)
+        except ValueError:
+            raise click.BadParameter("experiment run {} not found".format(run_id))
+    elif model_version_id:
+        try:
+            entity = client.get_registered_model_version(id=model_version_id)
+        except ValueError:
+            raise click.BadParameter("version {} not found".format(model_version_id))
+    else:
+        raise click.BadParameter("run-id or model_version_id should be specified")
```
Comment (convoliution) [inline client/verta/verta/_cli/deployment/download.py]:
```suggestion
        raise click.BadParameter("--run-id or --model-version-id should be specified")
```

File: client/verta/verta/_cli/deployment/download.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+    `verta registry download dockercontext BERT latest`
+
+    """
+    pass
+
+
+@download.command(name="dockercontext")
+@click.option("--run_id", "-r", help="Experiment Run id")
+@click.option("--model_version_id", "-m", help="Model Version id")
```
Comment (convoliution) [inline client/verta/verta/_cli/deployment/download.py]:
```suggestion
@click.option("--run-id", "-r", help="Experiment Run ID")
@click.option("--model-version-id", "-m", help="Model Version ID")
```

**Review by convoliution — CHANGES_REQUESTED** [review]

File: client/verta/tests/test_cli/test_endpoint.py (hunk 1 above)
Comment (evgeniilezhnin) [reply client/verta/tests/test_cli/test_endpoint.py]: replaced

File: client/verta/tests/test_cli/test_endpoint.py [hunk 4]
```diff
@@ -410,6 +411,56 @@ def test_update_autoscaling(self, client, created_endpoints, experiment_run, mod
...
+            filepaths = set(f.getnames())
+
+        assert "Dockerfile" in filepaths
+
+        np = pytest.importorskip("numpy")
+        from sklearn.linear_model import LogisticRegression
+
+        classifier = LogisticRegression()
+        classifier.fit(np.random.random((36, 12)), np.random.random(36).round())
+        model_version.log_model(classifier)
```
Comment (convoliution) [inline client/verta/tests/test_cli/test_endpoint.py]: We can use the same model object as before, provided by the fixture
```suggestion
        model_version.log_model(model_for_deployment['model'], custom_modules=[])
```

**Review by convoliution — APPROVED** [review]
One suggestion, then good to merge.

**Pull request merged**